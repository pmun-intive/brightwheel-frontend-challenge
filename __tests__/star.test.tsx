import { render, screen } from "@testing-library/react";
import Star from "@components/icon/star";

describe("icon", () => {
  it("renders a star svg with accessible name", () => {
    render(<Star />);

    const spinIcon = screen.getByRole("icon");

    expect(spinIcon).toHaveAccessibleName();
    expect(spinIcon).toMatchSnapshot();
  });

  it("renders a star svg with props width and height", () => {
    render(<Star height="250px" width="250px" />);

    const spinIcon = screen.getByRole("icon");

    expect(spinIcon).toHaveAttribute("width", "250px");
    expect(spinIcon).toHaveAttribute("height", "250px");
  });
});
