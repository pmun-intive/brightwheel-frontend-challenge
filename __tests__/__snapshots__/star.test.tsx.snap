// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon renders a star svg with accessible name 1`] = `
<svg
  aria-label="A star icon for your favorite results."
  height="50px"
  role="icon"
  viewBox="0 0 24 24"
  width="50px"
>
  <path
    d="M0 0h24v24H0z"
    fill="none"
  />
  <path
    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
    fill="none"
    stroke="black"
    stroke-width="1.5px"
  />
  <path
    d="M0 0h24v24H0z"
    fill="none"
  />
</svg>
`;
